File: Kelly/stack.linkedList/stack.linkedList.ts
class Node<T> {
  value: T;
  next: Node<T> | null;

  constructor(value: T) {
    this.value = value;
    this.next = null;
  }
}

class Stack<T> {
  private top: Node<T> | null;

  constructor() {
    this.top = null;
  }

  push(value: T): void {
    const newNode = new Node(value);
    newNode.next = this.top;
    this.top = newNode;
  }

  pop(): T | null {
    if (this.isEmpty()) {
      return null;
    }

    const value = this.top!.value;
    this.top = this.top!.next;
    return value;
  }

  peek(): T | null {
    return this.isEmpty() ? null : this.top!.value;
  }

  isEmpty(): boolean {
    return this.top === null;
  }

  clear(): void {
    this.top = null;
  }

  print(): void {
    let current = this.top;
    while (current !== null) {
      console.log(current.value);
      current = current.next;
    }
  }
}

File: Kelly/stack.linkedList/stack.linkedlist.cs
public class Node<T>
{
    public T Data { get; set; }
    public Node<T> Next { get; set; }

    public Node(T data)
    {
        Data = data;
        Next = null;
    }
}

public class Stack<T>
{
    private Node<T> top;
    private int count;

    public Stack()
    {
        top = null;
        count = 0;
    }

    public int Count
    {
        get { return count; }
    }

    public void Push(T item)
    {
        Node<T> newNode = new Node<T>(item);
        newNode.Next = top;
        top = newNode;
        count++;
    }

    public T Pop()
    {
        if (IsEmpty())
        {
            throw new InvalidOperationException("Stack is empty.");
        }

        T item = top.Data;
        top = top.Next;
        count--;
        return item;
    }

    public T Peek()
    {
        if (IsEmpty())
        {
            throw new InvalidOperationException("Stack is empty.");
        }

        return top.Data;
    }

    public bool IsEmpty()
    {
        return count == 0;
    }

    public void Clear()
    {
        top = null;
        count = 0;
    }
}
